torch/ao/quantization/fx/_model_report/_detector.py

```diff
@@ -0,0 +1,142 @@
+            backend_chosen
+        ]
+    else:
+        raise ValueError(
+            "Not configured to work with {}. Try a different default backend".format(
+                backend_chosen
+            )
+        )
+
+    # store information on submodules and if per_channel quantization is supported and used as well as qconfig information
+    per_channel_info = {"backend": backend_chosen, "per_channel_status": {}}
+
+    def _detect_per_channel_helper(module: nn.Module):
+        """
+        Recursive operation to determine if per_channel quantization is supported in modules and submodules.
+
+        Populates a dictionary in the higher level _detect_per_channel function.
+        Each entry maps the fully-qualified-name to information on whether per_channel quantization.
+
+        Args:
+            module: The current module that is being checked to see if it is per_channel qunatizable
+        """
+        # get the fully qualified name and check if in list of modules to include and list of modules to ignore
+        fqn = module_to_fqn(model, module)
+        is_in_include_list = (
+            True
+            if sum(list(map(lambda x: isinstance(module, x), supported_modules))) > 0
+            else False
+        )
+
+        # check if the module per_channel is supported
+        # based on backend
+        per_channel_supported = False
+
+        if is_in_include_list:
+            per_channel_supported = True
+
+            # assert statement for MyPy
+            q_config_file = module.qconfig
+            assert isinstance(q_config_file, QConfig)
+
+            # this object should either be fake quant or observer
+            q_or_s_obj = module.qconfig.weight.p.func()
+            assert isinstance(q_or_s_obj, FakeQuantize) or isinstance(
+                q_or_s_obj, ObserverBase
+            )
+
+            per_channel_used = False  # will be true if found in qconfig
+
+            if hasattr(
+                q_or_s_obj, "ch_axis"
+            ):  # then we know that per_channel quantization used
+
+                # all fake quants have channel axis so need to check is_per_channel
+                if isinstance(q_or_s_obj, FakeQuantize):
+                    if (
+                        hasattr(q_or_s_obj, "is_per_channel")
+                        and q_or_s_obj.is_per_channel
+                    ):
+                        per_channel_used = True
+                elif isinstance(q_or_s_obj, ObserverBase):
+                    # should be an observer otherwise
+                    per_channel_used = True
+                else:
+                    raise ValueError("Should be either observer or fake quant")
+
+            per_channel_info["per_channel_status"][fqn] = {
+                "per_channel_supported": per_channel_supported,
+                "per_channel_used": per_channel_used,
+            }
+
+        # recurse on children
+        for child in module.children():
```

Comment:
Instead of recursing on children, can we achieve the same result by using `module.named_modules` without recursion?

Reply:
I think they both get only the direct children, though named_modules gets name, child rather than just child.

Reply:
`named_modules` is a generator over all modules in the network, not just children of the current module. I'd recommend removing the recursion unless there is no way to do it without recursion.

Reply:
oh ya that makes sense